functionary/core/management/commands/utils.py

```diff
@@ -0,0 +1,29 @@
+from time import sleep
+from typing import Callable
+
+from django.db.utils import ProgrammingError
+
+SLEEP_DURATION = 2
+
+
+def run(command: Callable) -> None:
+    """Run the given the command
+
+    The given command will be executed to see if the necessary
+    database migrations have been run. If the migrations have not
+    been run, the command will be re-run after a delay.
+
+    Args:
+        command: The callable command to execute
+
+    Returns:
+        None
+    """
+    is_ready = False
+    while not is_ready:
+        try:
+            command()
+            is_ready = True
+        except ProgrammingError:
+            print("Migrations for the given command have not been run yet.")
```

Comment:
Two things with this:

1) Let's use the logger rather than a print statement.  Minor thing, but as is this ends up showing as a WARN rather than an ERROR, which you should be able to fix by calling `logger.error` instead.

2) Let's make the wording more general so that it applies to general database readiness issues:
  >Database is not ready. Ensure database is available and migrations have been run.  Retrying in 
  {SLEEP_DURATION}s.

You could break out the errors so that the message is specific to the actual problem, but the stack trace ends up being right above the error message anyway.

Reply:
Fixed